akka-remote/src/main/scala/akka/remote/artery/SystemMessageDelivery.scala

```diff
@@ -176,6 +185,10 @@ private[akka] class SystemMessageDelivery(
           case msg @ (_: SystemMessage | _: AckedDeliveryMessage) ⇒
             if (unacknowledged.size < maxBufferSize) {
               seqNo += 1
+              if (unacknowledged.isEmpty)
+                ackTimestamp = System.nanoTime()
```

Comment:
note that this is not when the buffer gets empty, it's when the buffer was empty